### PR TITLE
Improved YourCart section with new features

### LINKED ISSUE #74: Revamp the Shopping Cart screen
# Description
The shopping Cart screen looks very dull and the checkout button does not match the aesthetic of the app.
Also, the Your cart is empty pop-up looks plain

# Screenshot

### PR DESCRIPTION
**Pull request by Pranav108**
## Program
- [x] CodePeak
- [ ] Script Winter of Code
- [ ] None

## Issue
Fixes #74

## Description
If cart will be empty then UI will be a bit different and on clicking on `Add Item` user will be referred to Recommendation screen and if there will be any product in cart then we can see the option to purchase them.
Improved the designing of cart, button, appbar, etc.

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation change

## Screenshot
Screenshot of the fix if required.
drawingdrawingdrawing

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] The app runs properly
- [x] I have performed a self-review of my own code
- [x] I have formatted and commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by Saransh-cpp** [comment]
@Pranav108 can you also add screenshots for all the dialogue boxes?

**Comment by Pranav108** [comment]
I have Updated the screenShot in PR , that's the only dialog box I found on this screen . There was one more for the situation for cart to be empty but I removed that Instead of that now user will be referred to recommendation section on pression on `add item`.

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
@Pranav108, a quick review -

1. Please merge master and update your fork.
2. AppBar's color and the screen's color should be the same.
3. Please close the ad when a user switches to another screen.

**Comment by Pranav108** [comment]
I am unable to merge you master branch with my local branch , can just update the code for now?

**Comment by Pranav108** [comment]
> @Pranav108, a quick review -
> 
> 1. Please merge master and update your fork.
> 2. AppBar's color and the screen's color should be the same.
> 3. Please close the ad when a user switches to another screen.
> 
Now AppBar's color and the screen's color are be the same and ad will close when user switches to another screen. 
drawing

**Comment by Pranav108** [comment]
@Saransh-cpp do I need to add something more for this PR?

**Comment by Saransh-cpp** [comment]
Please resolve conflicts, honestly due to #92, it is getting a bit tough to review this PR but I will review it once the conflicts are resolved.

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
@Pranav108, apologies for taking forever to review. I am seeing an error on pressing the Add Item button -
```
The following assertion was thrown while handling a gesture:
'package:firebase_admob/firebase_admob.dart': Failed assertion: line 253 pos 12: '_allAds[id] != null': is not true.

When the exception was thrown, this was the stack: 
#2      MobileAd.dispose (package:firebase_admob/firebase_admob.dart:253:12)
#3      _YourOrdersState.build.<anonymous closure> (package:test_app/Screens/YourOrders.dart:284:39)
#4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:989:21)
#5      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:193:24)
#6      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:608:11)
#7      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:296:5)
#8      BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:267:7)
#9      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:157:27)
#10     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:444:20)
#11     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:420:22)
#12     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:278:11)
#13     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
#14     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
#15     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:296:7)
#16     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:279:7)
#20     _invoke1 (dart:ui/hooks.dart:185:10)
#21     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:293:7)
#22     _dispatchPointerDataPacket (dart:ui/hooks.dart:98:31)
```
I think you will have to check if the ad is even active and then dispose it off.

**Comment by Pranav108** [comment]
Now I am unable to run the code ,getting something like : 
`Your Flutter application is created using an older version of the Android embedding.`
` It is being deprecated in favor of Android embedding v2.`
So for running should I migrate it or ignore it for now?

**Comment by Saransh-cpp** [comment]
@Pranav108, you can work on this now.

**Comment by Pranav108** [comment]
> @Pranav108, apologies for taking forever to review. I am seeing an error on pressing the Add Item button -
> 
> ```
> The following assertion was thrown while handling a gesture:
> 'package:firebase_admob/firebase_admob.dart': Failed assertion: line 253 pos 12: '_allAds[id] != null': is not true.
> 
> When the exception was thrown, this was the stack: 
> #2      MobileAd.dispose (package:firebase_admob/firebase_admob.dart:253:12)
> #3      _YourOrdersState.build.<anonymous closure> (package:test_app/Screens/YourOrders.dart:284:39)
> #4      _InkResponseState._handleTap (package:flutter/src/material/ink_well.dart:989:21)
> #5      GestureRecognizer.invokeCallback (package:flutter/src/gestures/recognizer.dart:193:24)
> #6      TapGestureRecognizer.handleTapUp (package:flutter/src/gestures/tap.dart:608:11)
> #7      BaseTapGestureRecognizer._checkUp (package:flutter/src/gestures/tap.dart:296:5)
> #8      BaseTapGestureRecognizer.acceptGesture (package:flutter/src/gestures/tap.dart:267:7)
> #9      GestureArenaManager.sweep (package:flutter/src/gestures/arena.dart:157:27)
> #10     GestureBinding.handleEvent (package:flutter/src/gestures/binding.dart:444:20)
> #11     GestureBinding.dispatchEvent (package:flutter/src/gestures/binding.dart:420:22)
> #12     RendererBinding.dispatchEvent (package:flutter/src/rendering/binding.dart:278:11)
> #13     GestureBinding._handlePointerEventImmediately (package:flutter/src/gestures/binding.dart:374:7)
> #14     GestureBinding.handlePointerEvent (package:flutter/src/gestures/binding.dart:338:5)
> #15     GestureBinding._flushPointerEventQueue (package:flutter/src/gestures/binding.dart:296:7)
> #16     GestureBinding._handlePointerDataPacket (package:flutter/src/gestures/binding.dart:279:7)
> #20     _invoke1 (dart:ui/hooks.dart:185:10)
> #21     PlatformDispatcher._dispatchPointerDataPacket (dart:ui/platform_dispatcher.dart:293:7)
> #22     _dispatchPointerDataPacket (dart:ui/hooks.dart:98:31)
> ```
> 
> I think you will have to check if the ad is even active and then dispose it off.

Yeah I found this few days ago but app didn't crashes so I think we are okay with this.
With the existing `firebase_admob` package we can now fix this error with only one solution which is a small change in `package:firebase_admob/firebase_admob.dart`  file (with another warning ,because this file is read only), which may not be a good idea. 
And the second permanent solution will we use of `google_mobile_ads` because `firebase_admob` is deprecated.
image
I tried to implement `google_mobile_ads` package but got confused with different appID at several places.
So for ad improvement ,another issue will be effecient and I would love to give that another try.

**Comment by Saransh-cpp** [comment]
@Pranav108 can you create a new issue for this?

**Comment by Pranav108** [comment]
> @Pranav108 can you create a new issue for this?

Okay , I have raised an issue for that.
And apart from the current banner ads , in the new package we can also implement a optional full screen ads for few seconds. Should I ask to implement this extra feature in issue.

**Comment by Saransh-cpp** [comment]
> And apart from the current banner ads , in the new package we can also implement a optional full screen ads for few seconds. Should I ask to implement this extra feature in issue.

Yes, a new issue!

**Comment by Saransh-cpp** [comment]
@Pranav108 can you merge master?

**Comment by Pranav108** [comment]
My master into yours? No , coz I have a different branch for this issue.

**Comment by Saransh-cpp** [comment]
My master into your current branch

**Comment by Pranav108** [comment]
> My master into your current branch

yeah, I can merge there is no conflict now ,should I ?

**Comment by Saransh-cpp** [comment]
> yeah, I can merge there is no conflict now ,should I ?

Yes, your branch is 22 commits behind, would be difficult to review.

**Comment by Saransh-cpp** [comment]
@Pranav108, I am really sorry for the delay but now that the bug associated with `YourOrdrs.dart` has been solved, can you merge my master branch again? I'll review as soon as you merge!

**Comment by Pranav108** [comment]
> @Pranav108, I am really sorry for the delay but now that the bug associated with `YourOrdrs.dart` has been solved, can you merge my master branch again? I'll review as soon as you merge!

It's okay bro, I've merged with your master branch.

File: lib/Screens/YourOrders.dart [hunk 1]
```diff
@@ -61,84 +61,147 @@ class _YourOrdersState extends State<YourOrders> {
...
         home: SafeArea(
             child: Scaffold(
           appBar: AppBar(
-            iconTheme: IconThemeData(color: Colors.black),
-            backgroundColor: Colors.white,
+            iconTheme: IconThemeData(
+              color: Colors.pink,
+            ),
+            backgroundColor: Colors.pink[50],
             elevation: 1,
```
Comment (Saransh-cpp) [inline lib/Screens/YourOrders.dart:75]:
```suggestion
            elevation: 0,
```

File: lib/Screens/YourOrders.dart [hunk 2]
```diff
@@ -61,84 +61,147 @@ class _YourOrdersState extends State<YourOrders> {
...
-                                  height: 120,
-                                  width: 140,
-                                  fit: BoxFit.fill,
+              : isCartEmpty
+                  ? Column(
+                      mainAxisAlignment: MainAxisAlignment.spaceAround,
+                      crossAxisAlignment: CrossAxisAlignment.center,
+                      children: [
+                        Image.asset(
+                          'images/emptycart.png',
```
Comment (Saransh-cpp) [inline lib/Screens/YourOrders.dart:90]: This gives me the following error -



I think you forgot to add the file in `pubspec.yaml`?

**Review by Saransh-cpp — CHANGES_REQUESTED** [review]
Thanks, @Pranav108! Awesome work! The ad still gives the same error but I think it would be taken care of in #111. A couple of minor comments below -

**Review by Saransh-cpp — APPROVED** [review]

**Comment by Pranav108** [comment]
Thanks bro🤩

**Pull request merged**